### DISCUSSION TITLE
Create custom theme?

### DISCUSSION THREAD

**Discussion by steveiliop56** [cockpit-project/cockpit#19570] [Q&A] [answered]
Hello,

I want to make a custom cockpit theme for a custom raspberry pi os that I am building (based on bookworm) and I am thinking is there any way to add a custom theme to the files of cockpit and like replace a default or is it a matter of building the package?

**Comment by martinpitt** [comment, marked as answer]
You have to rebuild the package for that. We don't support "themes", so there's no tried-and-tested/documented mechanism for that, but if you just want to change a few colors here and there, you can of course hack the SCSS files.

**Reply by steveiliop56** [reply]
Yeah that's what I want to change only colors. Were can I find the file?

**Reply by steveiliop56** [reply]
Oh nvm found it.

**Reply by steveiliop56** [reply]
Oh noo sorry I was the "found it" but no explanation guy. Man sorry I don't remember.

**Reply by steveiliop56** [reply]
After a quick look the only css files I can find in cockpit are:

```
/usr/share/cockpit/file-sharing/assets/index-GjkWr_uJ.css
/usr/share/cockpit/static/login.css
/usr/share/cockpit/branding/default/branding.css
/usr/share/cockpit/branding/debian/branding.css
/usr/share/cockpit/branding/registry/branding.css
/usr/share/cockpit/branding/arch/branding.css
/usr/share/cockpit/branding/kubernetes/branding.css
```

I think the css is somehow bundled.

**Comment by Icosa-Consulting** [comment]
I did this for the login page, however when APT updated cockpit to 3.10 it reset all my customizations back to default. I noticed that the login "theme" is broken down by subdirectory, in my case /usr/share/cockpit/branding/debian/ I'm wondering if another directory could be utilized to have a custom branding package?

**Reply by martinpitt** [reply]
Unfortunately cockpit-ws doesn't respect /usr/local/share/branding/ -- that would be the obvious place how to override them. You really shouldn't write /usr/share/cockpit, that's package manager territory.
